spectrum-offchain-lm/src/data/bundle.rs

```diff
@@ -203,14 +209,24 @@ impl TryFromBox for StakingBundle {
                     .ok()?,
                 );
                 let vlq = tokens.get(0)?.clone();
-                let tmp = tokens.get(1)?.clone();
-                let bundle_key = tokens.get(2)?.clone();
+                let (tmp, bundle_key) = if tokens.len() == 3 {
+                    let tmp = TypedAssetAmount::from_token(tokens.get(1)?.clone());
+                    let bundle_key = tokens.get(2)?.clone();
+                    (tmp, bundle_key)
+                } else {
+                    let bundle_key = tokens.get(1)?.clone();
+
+                    // No TMP tokens and we don't have the TokenId on hand, just generate a dummy
+                    // value.
+                    let tmp = TypedAssetAmount::new(force_any_val::<TokenId>(), 0);
+                    (tmp, bundle_key)
```

Comment:
This should be reflected in the type signature instead of making up a fake value.

Reply:
Wrap the type in an `Option`.